File: src/com/wang/test/gui/javaMain.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.wang.test.gui;

import com.wang.Game2dEngine.director.Director;
import com.wang.Game2dEngine.scene.Scene;
import com.wang.common.JavaTheme;
import com.wang.test.collision.CollisionScene;
import com.wang.test.fireworks.FireworksScene;
import com.wang.test.fountain.FountainScene;
import com.wang.test.galaxies.GalaxiesDemoScene;
import com.wang.test.navigate.NavigateScene;
import com.wang.test.universe.GravityDemoScene;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.awt.event.WindowEvent;
import java.awt.event.WindowListener;
import static java.lang.System.exit;
import javax.swing.JFrame;
import javax.swing.JMenu;
import javax.swing.JMenuBar;
import javax.swing.JMenuItem;
import javax.swing.JPanel;
import javax.swing.SwingUtilities;

/**
 *
 * @author ricolwang
 */
public class javaMain
{

    JMenu menuGame, menuOption;
    JMenuItem menuFountain, menuFireworks, menuCollision, menuRocket, menuUniverse, menuGalaxies, menuSeperator, menuExit, menuPause;
    JPanel panelGame = new JPanel();

    public javaMain()
    {
        JFrame f = new JFrame("Game2dEngine Test");
        JMenuBar mb = new JMenuBar();

        menuGame = new JMenu("Game");

        menuFountain = new JMenuItem("Fountan");
        menuFountain.addActionListener(new ActionListener()
        {
            @Override
            public void actionPerformed(ActionEvent e)
            {
                Director.getSharedInstance().showScene(new FountainScene());
            }
        });

        menuFireworks = new JMenuItem("Fireforks");
        menuFireworks.addActionListener(new ActionListener()
        {
            @Override
            public void actionPerformed(ActionEvent e)
            {
                Director.getSharedInstance().showScene(new FireworksScene());
            }
        });

        menuCollision = new JMenuItem("Collision");
        menuCollision.addActionListener(new ActionListener()
        {
            @Override
            public void actionPerformed(ActionEvent e)
            {
                Director.getSharedInstance().showScene(new CollisionScene());
            }
        });

        menuRocket = new JMenuItem("Rocket");
        menuRocket.addActionListener(new ActionListener()
        {
            @Override
            public void actionPerformed(ActionEvent e)
            {
                Director.getSharedInstance().showScene(new NavigateScene());
            }
        });

        menuUniverse = new JMenuItem("Universe");
        menuUniverse.addActionListener(new ActionListener()
        {
            @Override
            public void actionPerformed(ActionEvent e)
            {
                Director.getSharedInstance().showScene(new GravityDemoScene());
            }
        });

        menuGalaxies = new JMenuItem("Galaxies");
        menuGalaxies.addActionListener(new ActionListener()
        {
            @Override
            public void actionPerformed(ActionEvent e)
            {
                Director.getSharedInstance().showScene(new GalaxiesDemoScene());
            }
        });

        menuSeperator = new JMenuItem("-");

        menuExit = new JMenuItem("Exit");
        menuExit.addActionListener(new ActionListener()
        {
            @Override
            public void actionPerformed(ActionEvent e)
            {
                exit(0);
            }
        });

        menuGame.add(menuFountain);
        menuGame.add(menuFireworks);
        menuGame.add(menuCollision);
        menuGame.add(menuRocket);
        menuGame.add(menuUniverse);
        menuGame.add(menuGalaxies);
        menuGame.add(menuSeperator);
        menuGame.add(menuExit);

        menuOption = new JMenu("Option");

        menuPause = new JMenuItem("Pause");
        menuPause.addActionListener(new ActionListener()
        {
            @Override
            public void actionPerformed(ActionEvent e)
            {
                Scene theScene = Director.getSharedInstance().currentScene;
                if (theScene != null) theScene.pause();
            }
        });
        menuOption.add(menuPause);

        mb.add(menuGame);
        mb.add(menuOption);

        f.setContentPane(panelGame);
        f.setJMenuBar(mb);
        f.setSize(800, 600);
        f.setLayout(null);
        f.setVisible(true);
        f.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);

        f.addWindowListener(new WindowListener()
        {
            @Override
            public void windowOpened(WindowEvent e)
            {
                Director.getSharedInstance().setParent(panelGame);
            }

            @Override
            public void windowClosing(WindowEvent e)
            {
            }

            @Override
            public void windowClosed(WindowEvent e)
            {
            }

            @Override
            public void windowIconified(WindowEvent e)
            {
            }

            @Override
            public void windowDeiconified(WindowEvent e)
            {
            }

            @Override
            public void windowActivated(WindowEvent e)
            {
            }

            @Override
            public void windowDeactivated(WindowEvent e)
            {
            }
        });
    }

    public static void main(String[] args)
    {
        SwingUtilities.invokeLater(new Runnable()
        {
            public void run()
            {
                JavaTheme.setLookAndFeel(JavaTheme.LOOKANDFEEL_NIBUM);
                new javaMain();
            }
        });
    }
}

File: src/com/wang/test/gui/javaMain.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.wang.test.gui

import com.wang.Game2dEngine.director.Director
import com.wang.common.JavaTheme
import com.wang.test.collision.CollisionScene
import com.wang.test.fireworks.FireworksScene
import com.wang.test.fountain.FountainScene
import com.wang.test.galaxies.GalaxiesDemoScene
import com.wang.test.navigate.NavigateScene
import com.wang.test.universe.GravityDemoScene
import java.awt.event.WindowEvent
import java.awt.event.WindowListener
import javax.swing.*

/**
 * @author ricolwang
 */
class javaMain
{
    var menuGame: JMenu
    var menuOption: JMenu
    var menuFountain: JMenuItem
    var menuFireworks: JMenuItem
    var menuCollision: JMenuItem
    var menuRocket: JMenuItem
    var menuUniverse: JMenuItem
    var menuGalaxies: JMenuItem
    var menuSeperator: JMenuItem
    var menuExit: JMenuItem
    var menuPause: JMenuItem
    var panelGame = JPanel()

    companion object
    {
        @JvmStatic
        fun main(args: Array<String>)
        {
            SwingUtilities.invokeLater {
                JavaTheme.setLookAndFeel(JavaTheme.LOOKANDFEEL_NIBUM)
                javaMain()
            }
        }
    }

    init
    {
        val f = JFrame("Game2dEngine Test")
        val mb = JMenuBar()
        menuGame = JMenu("Game")
        menuFountain = JMenuItem("Fountan")
        menuFountain.addActionListener { Director.sharedInstance.showScene(FountainScene()) }
        menuFireworks = JMenuItem("Fireforks")
        menuFireworks.addActionListener { Director.sharedInstance.showScene(FireworksScene()) }
        menuCollision = JMenuItem("Collision")
        menuCollision.addActionListener { Director.sharedInstance.showScene(CollisionScene()) }
        menuRocket = JMenuItem("Rocket")
        menuRocket.addActionListener { Director.sharedInstance.showScene(NavigateScene()) }
        menuUniverse = JMenuItem("Universe")
        menuUniverse.addActionListener { Director.sharedInstance.showScene(GravityDemoScene()) }
        menuGalaxies = JMenuItem("Galaxies")
        menuGalaxies.addActionListener { Director.sharedInstance.showScene(GalaxiesDemoScene()) }
        menuSeperator = JMenuItem("-")
        menuExit = JMenuItem("Exit")
        menuExit.addActionListener { System.exit(0) }
        menuGame.add(menuFountain)
        menuGame.add(menuFireworks)
        menuGame.add(menuCollision)
        menuGame.add(menuRocket)
        menuGame.add(menuUniverse)
        menuGame.add(menuGalaxies)
        menuGame.add(menuSeperator)
        menuGame.add(menuExit)
        menuOption = JMenu("Option")
        menuPause = JMenuItem("Pause")
        menuPause.addActionListener {
            val theScene = Director.sharedInstance.currentScene
            theScene?.pause()
        }
        menuOption.add(menuPause)
        mb.add(menuGame)
        mb.add(menuOption)
        f.contentPane = panelGame
        f.jMenuBar = mb
        f.setSize(800, 600)
        f.layout = null
        f.isVisible = true
        f.defaultCloseOperation = JFrame.EXIT_ON_CLOSE
        f.addWindowListener(object : WindowListener
        {
            override fun windowOpened(e: WindowEvent)
            {
                Director.sharedInstance.setParent(panelGame)
            }

            override fun windowClosing(e: WindowEvent)
            {
            }

            override fun windowClosed(e: WindowEvent)
            {
            }

            override fun windowIconified(e: WindowEvent)
            {
            }

            override fun windowDeiconified(e: WindowEvent)
            {
            }

            override fun windowActivated(e: WindowEvent)
            {
            }

            override fun windowDeactivated(e: WindowEvent)
            {
            }
        })
    }
}